imagepipeline-base/src/main/java/com/facebook/imageformat/ImageFormatCheckerUtils.java

```diff
@@ -59,5 +59,31 @@ public static boolean startsWithPattern(
     return true;
   }
 
+  /**
+   * Checks if byteArray interpreted as sequence of bytes contains the pattern.
+   * @param byteArray the byte array to be checked
+   * @param pattern the pattern to check
+   * @return index of beginning of pattern, if found; otherwise -1
+   */
+  public static int containsPattern(
```

Comment:
Gotcha. I had originally wanted to follow the style of the startsWithPattern() method above this one, but if you don't think it's necessary here, then that's fine with me. Also, can't we consider the time complexity to be O(n), or at least O(mn), since the length of the pattern is constant? I was sort of following the [String.contains()](http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/lang/String.java#String.indexOf%28char%5B%5D%2Cint%2Cint%2Cchar%5B%5D%2Cint%2Cint%2Cint%29) algorithm, because it does basically the same thing.